subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -508,12 +595,59 @@ static struct tls_context *tls_clone(struct tls_context *source_tls)
 	memcpy(&target_tls->options, &source_tls->options,
 	       sizeof(target_tls->options));
 
+#if defined(CONFIG_WOLFSSL)
+    if (NULL != source_tls->host_name)
```

Comment:
Are we not able to create a wolfSSL CTX and SSL object at this point to set them directly? Seems odd to have to malloc and copy the settings to use it later.

Reply:
I was also not thrilled about this flow but it doesnt seem so. I dont know until connect/accept whether its a client or server. A common flow seems to be they create a socket and then setsockopt, so I need to be able to handle it. mbedtls utilizes the same flow.